### PR TITLE
Add an upgrade nightly job for Katello

### PR DESCRIPTION
**Pull request by zjhuntin**

### CODE REVIEW

**Comment by zjhuntin** [comment]
Requires: https://github.com/theforeman/forklift/pull/795

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/katelloUpgradeNightly.groovy [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    stages {
+        stage('Setup Environment') {
+            steps {
+                deleteDir()
+                git url: 'https://github.com/theforeman/forklift.git'
+            }
+        }
+        stage('Provision Node') {
+            steps {
+                dir('foreman-infra') {
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/katelloUpgradeNightly.groovy]: I thought provision cloned this repository for us?

File: puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-upgrade-nightly.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+- job:
+    name: 'foreman-upgrade-nightly'
+    project-type: 'pipeline'
+    sandbox: true
+    dsl:
+      !include-raw:
+        - pipelines/test/katelloUpgradeNightly.groovy
+        - ../theforeman.org/pipelines/lib/duffy.groovy
+        - ../theforeman.org/pipelines/lib/ansible.groovy
+        - ../theforeman.org/pipelines/lib/folderChanged.groovy
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-upgrade-nightly.yml]: Is there functionality in here used?

**Comment by ehelms** [comment]
@zjhuntin This looks like it needs a rebase

**Comment by zjhuntin** [comment]
@ehelms rebased, sorry!

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/lib/foremanCentosJob.groovy [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                dir('foreman-infra') {
+                    git url: 'https://github.com/theforeman/foreman-infra.git'
+                }
+
+                provision()
+            }
+        }
+        stage('Install Pipeline Requirements') {
+            steps {
+                runPlaybook('playbooks/setup_forklift.yml', cico_inventory('./'), [], ['-b'])
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/centos.org/pipelines/lib/foremanCentosJob.groovy]: This method signature changed so you'll need to update each call to runPlaybook

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/lib/foremanCentosJob.groovy [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            steps {
+                deleteDir()
+                git url: 'https://github.com/theforeman/forklift.git'
+            }
+        }
+        stage('Provision Node') {
+            steps {
+                dir('foreman-infra') {
+                    git url: 'https://github.com/theforeman/foreman-infra.git'
+                }
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/centos.org/pipelines/lib/foremanCentosJob.groovy]: The provision method does this for you so you can omit it here.

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/katelloUpgradeNightly.groovy [hunk 5]
```diff
@@ -0,0 +1,4 @@
+def pipelineVars() {
+    vars = ['boxes': ['pipeline-upgrade-centos7'], 'pipeline': 'pipeline_upgrade_nightly.yml']
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/katelloUpgradeNightly.groovy]: Maybe we make this more generic and just call it playbook?

**Comment by zjhuntin** [comment]
@ehelms Updated

File: puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-upgrade-nightly.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+- job:
+    name: 'foreman-upgrade-nightly'
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-upgrade-nightly.yml]: This name and file need to change to match the job

**Pull request merged**